setenv.pl

```diff
@@ -6,9 +6,20 @@
 my $root;
 
 BEGIN {
-    use File::Basename qw(dirname);
-    use File::Spec;
-    $root = dirname(File::Spec->rel2abs(__FILE__));
+    use Cwd qw(abs_path);
+    use File::Basename qw(dirname basename);
+    $root = dirname(abs_path(__FILE__));
+}
+
+# Check there is not a later timestamped deploy, if running in such a system
+
+if ($root =~ /fixmystreet-\d\d\d\d-\d\d-\d\dT\d\d-\d\d-\d\d$/) {
+    my @deploys = sort map { basename $_ } glob("$root/../fixmystreet-*");
```

Comment:
I am not sure what is going on but this use of glob upsets at least the test running scripts:

```
photek:fixmystreet struan$ (2079-check-latest-deploy $%=) ./script/test
Spinning up a Pg cluster/database...
Use of uninitialized value in glob at /Users/struan/src/mysociety/FMS-vagrant/fixmystreet/local/lib/perl5/Test/PostgreSQL.pm line 72.
Use of uninitialized value in glob at /Users/struan/src/mysociety/FMS-vagrant/fixmystreet/local/lib/perl5/Test/PostgreSQL.pm line 72.
Use of uninitialized value in glob at /Users/struan/src/mysociety/FMS-vagrant/fixmystreet/local/lib/perl5/Test/PostgreSQL.pm line 72.
```
which seems to be because caused by this changing the `glob` behaviour when called on a non-existant path.

I don't totally understand what is going on though.

Reply:
If you remove the glob here then it all works fine.

Reply:
Yeah, using `glob` autoloads File::Glob but then the unload bit was removing it (it's for removing File::Spec etc) which would confuse poor `glob`! Fixed to skip glob.